analyzers/src/SonarAnalyzer.Common/Json/Parsing/LexicalAnalyzer.cs

```diff
@@ -326,15 +326,15 @@ private object ReadNumberValue()
             object BuildResult()
             {
                 var baseValue = @decimal == null
-                    ? (object)int.Parse(integer.ToString())
-                    : decimal.Parse(integer + CultureInfo.CurrentCulture.NumberFormat.NumberDecimalSeparator + @decimal);
+                    ? (object)double.Parse(@double.ToString())
+                    : decimal.Parse(@double + CultureInfo.CurrentCulture.NumberFormat.NumberDecimalSeparator + @decimal);
```

Comment:
I would avoid the use of `CurrentCulture`:
```cs
decimal.Parse(integral + '.'+ @decimal, CultureInfo.InvariantCulture);
```

Reply:
If you change `CultureInfo.CurrentCulture.NumberFormat.NumberDecimalSeparator` into `CultureInfo.InvariantCulture.NumberFormat.NumberDecimalSeparator` you also need to pass the same `InvariantCulture` to the `double.Parse` and to the `decimal.Parse` methods.

Otherwise, the parse won't work on machines having a locale with a decimal separator different from `.` (e.g. French or Italian).

So the two lines should become:
```cs
? (object)double.Parse(@double.ToString(), CultureInfo.InvariantCulture)
: decimal.Parse(@double + CultureInfo.CurrentCulture.NumberFormat.NumberDecimalSeparator + @decimal, CultureInfo.InvariantCulture);
```